compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRTypeWriter.java

```diff
@@ -52,18 +53,22 @@
 import org.wso2.ballerinalang.compiler.semantics.model.types.BXMLType;
 import org.wso2.ballerinalang.compiler.util.TypeTags;
 
+import java.util.LinkedList;
+
 /**
  * Writes bType to a Byte Buffer in binary format.
  * A ConstPool is used to store string typed information.
  * 
  * @since 0.995.0
  */
 public class BIRTypeWriter implements TypeVisitor {
+    public static final int TYPE_TAG_SELF = 50;
     private final ByteBuf buff;
 
     private final ConstantPool cp;
     private final BMapType anydataMapType =
```

Comment:
We can remove this too right, since we do not have to overwrite the error detail now?

Reply:
Thanks, fixed in #14885